src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -1036,6 +1050,10 @@ def populate_column_dependency(
                     )
                 else:  # A referenced column can't be found. The query has an error.
                     raise DeriveColumnDependencyError()
+    else:
+        raise ValueError(
+            f"Unexpected dependent_column_names: {dependent_column_names}"
+        )  # pragma: no cover
```

Comment:
I think this is breaking the tests. We might need something line `elif dependent_column_names is not None` at line 1043

Reply:
Pushed the fix to my branch